python/cudf/cudf_pandas_tests/test_cudf_pandas.py

```diff
@@ -1979,3 +1979,18 @@ def test_numpy_data_access():
     actual = xs.values.data
 
     assert type(expected) is type(actual)
+
+
+def test_pickle_round_trip_proxy_numpy_array(array):
+    arr, proxy_arr = array
+    pickled_arr = BytesIO()
+    pickled_proxy_arr = BytesIO()
+    pickle.dump(arr, pickled_arr)
+    pickle.dump(proxy_arr, pickled_proxy_arr)
+
+    pickled_arr.seek(0)
+    pickled_proxy_arr.seek(0)
+
+    np.testing.assert_equal(
+        pickle.load(pickled_proxy_arr), pickle.load(pickled_arr)
+    )
```

Comment:
TODO: This can be simplified a bit, I think. I'll tack it on to another PR.
```python
def test_pickle_round_trip_proxy_numpy_array(array):
    arr, proxy_arr = array
    np.testing.assert_equal(
        pickle.loads(pickle.dumps(proxy_arr)), 
        pickle.loads(pickle.dumps(arr))
    )
```